src/NBitcoin/ConsensusOptions.cs

```diff
@@ -106,12 +106,39 @@ public ConsensusOptions()
     /// </summary>
     public class PosConsensusOptions : ConsensusOptions
     {
-        /// <summary>Coinstake minimal confirmations softfork activation height for the mainnet.</summary>
+        /// <summary>Coinstake minimal confirmations softfork activation height for mainnet.</summary>
         public const int CoinstakeMinConfirmationActivationHeightMainnet = 1005000;
 
-        /// <summary>Coinstake minimal confirmations softfork activation height for the testnet.</summary>
+        /// <summary>Coinstake minimal confirmations softfork activation height for testnet.</summary>
         public const int CoinstakeMinConfirmationActivationHeightTestnet = 436000;
 
+        /// <summary>
+        /// Proven headers activation height for mainnet.
+        /// TODO: Set it to the real value once it is known.
+        /// </summary>
+        public const int ProvenHeadersActivationHeightMainnet = 20_000_000;
```

Comment:
Lets go over again why we need activation?

We should just know that a peer has or has not PH and if it has we ask PH from it otherwise we don't.
Eventually when no more X nodes we will only use PH.

I guess the issue is that then an attacker can just not send PH and fake blocks so at some point we can push in the code a condition that we need to have at least 50% outbound as PH nodes

Reply:
after activation heighr node will require proven headers

Reply:
>Eventually when no more X nodes we will only use PH.

Node will use normal headers for checkpointed blocks since we don't need to validate those headers.